filter/store.go

```diff
@@ -0,0 +1,153 @@
+	return &Store{
+		filterAPIURL: filterAPIURL,
+		httpClient:   httpClient,
+	}
+}
+
+// HTTPClient dependency.
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+// PutCSVData allows the filtered file data to be sent back to the filter store when complete.
+func (store *Store) PutCSVData(filterJobID string, url string, size int64) error {
+
+	// todo: make call to the filter API.
+
+	return nil
+}
+
+// GetFilter returns filter data from the filter API for the given ID
+func (store *Store) GetFilter(filterJobID string) (*observation.Filter, error) {
+
+	filter, err := store.getFilterMetaData(filterJobID)
+	if err != nil {
+		return nil, err
+	}
+
+	dimensionList, err := store.getFilterDimensionList(filter)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, dimension := range dimensionList {
+
+		dimensionFilter, err := store.getFilterDimensionOptions(dimension.URL)
+		if err != nil {
+			return nil, err
+		}
+
+		filter.DimensionFilters = append(filter.DimensionFilters, dimensionFilter)
+	}
+
+	return filter, nil
+}
+
+// call the filter API for filter meta data.
+func (store *Store) getFilterMetaData(filterJobID string) (*observation.Filter, error) {
+
+	url := store.filterAPIURL + "/filters/" + filterJobID
+
+	bytes, err := store.makeRequest("GET", url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var filter *observation.Filter
+	JSONErr := json.Unmarshal(bytes, &filter)
+	if JSONErr != nil {
+		return nil, JSONErr
+	}
+
+	return filter, nil
+}
+
+// call the filter API for filter meta data.
+func (store *Store) getFilterDimensionOptions(dimensionURL string) (*observation.DimensionFilter, error) {
+
+	dimensionOptionsURL := dimensionURL + "/options"
+
+	bytes, err := store.makeRequest("GET", dimensionOptionsURL, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var dimensionFilter *observation.DimensionFilter
+	err = json.Unmarshal(bytes, &dimensionFilter)
+	if err != nil {
+		return nil, err
+	}
+
+	return dimensionFilter, nil
+}
+
+func (store *Store) getFilterDimensionList(filter *observation.Filter) ([]*observation.DimensionFilter, error) {
+
+	bytes, err := store.makeRequest("GET", filter.DimensionListURL, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var dimensionURLs []*observation.DimensionFilter
+	err = json.Unmarshal(bytes, &dimensionURLs)
+	if err != nil {
+		return nil, err
+	}
+
+	return dimensionURLs, nil
+}
+
+// common function for handling a HTTP request and response codes.
+func (store *Store) makeRequest(method, url string, body io.Reader) ([]byte, error) {
+
+	request, err := http.NewRequest(method, url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	response, responseError := store.httpClient.Do(request)
+	if responseError != nil {
+		return nil, responseError
+	}
+
+	switch response.StatusCode {
+	case http.StatusNotFound:
+		return nil, ErrFilterJobNotFound
+	case http.StatusInternalServerError:
+		return nil, ErrFilterAPIError
```

Comment:
I have added a default case and returned an 'unrecognised' error. I also had to add a case for a success response, to prevent it falling into the default.